aiopg/utils.py

```diff
@@ -86,7 +86,20 @@ class _SAConnectionContextManager(_ContextManager):
     if PY_35:  # pragma: no branch
         if PY_352:
             def __aiter__(self):
-                return self._coro
+                return self
+
+            @asyncio.coroutine
+            def __anext__(self):
+                if self._obj is None:
+                    self._obj = yield from self._coro
+
+                try:
+                    return (yield from self._obj.__anext__())
+                except StopAsyncIteration:
+                    self._obj.close()
+                    self._obj = None
+                    raise
```

Comment:
Ideally, we would just make `__aiter__` an async generator so we don't have to supply this `__anext__` proxying code, but this would break for the small handful of CPython releases >=3.5.2 and <3.6